### PR TITLE
test: pill

### PR DESCRIPTION
**Pull request by DianeGit22**
### Proposed behaviour

Refactor for pill component to use new cypress-component-react framework for testing.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `pager.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

File: src/components/pill/pill.test.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import * as React from "react";
+import Pill from "./pill.component";
+import { pillPreview } from "../../../cypress/locators/pill/index";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import { closeIconButton } from "../../../cypress/locators/index";
+
+const specialCharacters = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const testData = "cypress_data";
+
+// eslint-disable-next-line react/prop-types
```
Comment (DlgSHi) [inline src/components/pill/pill.test.js:10]: do we need this comment here?

File: src/components/pill/pill.test.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      "should render Pill component with colorVariant set to %s",
+      (color, output) => {
+        CypressMountWithProviders(
+          <Pill pillRole="status" colorVariant={color}>
+            Pill
+          </Pill>
+        );
+
+        pillPreview()
+          .should("have.css", "border", output)
```
Comment (DlgSHi) [inline src/components/pill/pill.test.js:47]: like this: 
```suggestion
          .should("have.css", "border", `2px solid ${output}`)
```

File: src/components/pill/pill.test.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      "should render Pill component with color fill to %s",
+      (color, output) => {
+        CypressMountWithProviders(
+          <Pill pillRole="status" colorVariant={color} fill>
+            Pill
+          </Pill>
+        );
+
+        pillPreview()
+          // eslint-disable-next-line prefer-template
```
Comment (DlgSHi) [inline src/components/pill/pill.test.js:77]: why this comment is here ?

File: src/components/pill/pill.test.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+          .should("have.css", "border", "2px solid " + output)
+          .should("have.css", "background-color", output);
+      }
+    );
+
+    it("should call the delete action after the CloseIcon is clicked", () => {
```
Comment (DlgSHi) [inline src/components/pill/pill.test.js:83]: let's move the `event` testing into a separate `describe`. And than just use `callback` initialisation only once in `beforeEach`

File: src/components/pill/pill.test.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+      pillPreview()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    // it.each(["tag", "status"])(
```
Comment (DlgSHi) [inline src/components/pill/pill.test.js:109]: do we need this commented code?

File: src/components/pill/pill.test.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    //   "should render Dialog component with pillRole set to %s",
+    //   (role) => {
+    //     CypressMountWithProviders(<PillComponent pillRole={role} />);
+
+    //     pillPreview().should("have.attr", "pillRole", role);
+    //   }
+    // );
+
+    it.each([
+      ["--colorsSemanticCaution500", "2px solid rgb(239, 103, 0)"],
```
Comment (DlgSHi) [inline src/components/pill/pill.test.js:119]: the same is here with `2px solid`

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: src/components/pill/pill.test.js (hunk 1 above)
Comment (DianeGit22) [reply src/components/pill/pill.test.js]: removed

File: src/components/pill/pill.test.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+      "should render Pill using %s as label",
+      (label) => {
+        CypressMountWithProviders(<Pill>{label}</Pill>);
+
+        pillPreview().should("have.text", label);
+      }
+    );
+
+    it.each([
+      ["warning", "2px solid rgb(242, 133, 51)"],
```
[the remark this answers is not in the document]
Comment (DianeGit22) [reply src/components/pill/pill.test.js]: Done

File: src/components/pill/pill.test.js (hunk 2 above)
Comment (DianeGit22) [reply src/components/pill/pill.test.js]: done

File: src/components/pill/pill.test.js (hunk 3 above)
Comment (DianeGit22) [reply src/components/pill/pill.test.js]: removed

File: src/components/pill/pill.test.js (hunk 4 above)
Comment (DianeGit22) [reply src/components/pill/pill.test.js]: Done

File: src/components/pill/pill.test.js (hunk 5 above)
Comment (DianeGit22) [reply src/components/pill/pill.test.js]: Test is passing now

File: src/components/pill/pill.test.js (hunk 6 above)
Comment (DianeGit22) [reply src/components/pill/pill.test.js]: Done

**Review by DlgSHi — APPROVED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]
Tests look all good to me, I've made some amendment to the description to follow the pattern. Put in some details for proposed/current behaviour and testing instructions.

**Review by stephenogorman — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**